### PR TITLE
Clean dirs in data termination job

### PR DESCRIPTION
**Pull request by boyan-velinov**
Jira: LMCROSSITXSADEPLOY-1162

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/security/data/termination/DataTerminationService.java [hunk 1]
```diff
@@ -53,6 +59,15 @@ public void deleteOrphanUserData() {
...
             deleteUserOperationsOrphanData(spaceId);
+            deleteSpaceLeftovers(spaceId);
+        }
+    }
+
+    private void deleteSpaceLeftovers(String spaceId) {
+        try {
+            fileService.deleteSpaceLeftovers(spaceId);
+        } catch (FileStorageException e) {
+            throw new SLException(e, "Deletion of space leftovers failed");
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/security/data/termination/DataTerminationService.java]: Extract this: "Deletion of space leftovers failed" to a constant :)

File: com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/AbstractFileService.java [hunk 2]
```diff
@@ -240,6 +240,8 @@ public Integer execute(Connection connection) throws SQLException {
             throw new FileStorageException(e.getMessage(), e);
         }
     }
+    
+    public abstract void deleteSpaceLeftovers(final String space) throws FileStorageException;
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/AbstractFileService.java]: Isn't this the spaceId?

File: com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/FileSystemFileService.java [hunk 3]
```diff
@@ -106,6 +108,18 @@ private Path getFilePath(FileEntry entry) throws IOException {
     public int deleteAll(String space, String namespace) throws FileStorageException {
         throw new UnsupportedOperationException();
     }
+    
+    @Override
+    public void deleteSpaceLeftovers(String space) throws FileStorageException {
+        File spaceDirectory = getSpaceDirectory(space).toFile();
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/FileSystemFileService.java]: Why not using Path?

File: com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/FileSystemFileService.java [hunk 4]
```diff
@@ -106,6 +108,18 @@ private Path getFilePath(FileEntry entry) throws IOException {
...
+        try {
+            if (spaceDirectory.exists()) {
+                FileUtils.deleteDirectory(spaceDirectory);
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/FileSystemFileService.java]: Why not using Files API? What does FileUtils provide more?

File: com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/FileSystemFileService.java (hunk 3 above)
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/FileSystemFileService.java]: I believe the getSpaceDirectory brings some duplication here.
See:

>    private Path getSpaceDirectory(String space) {
>         return Paths.get(storagePath, space);
>     }

And this:

> private Path getFilesPerSpaceDirectory(String space) {
>     return Paths.get(storagePath, space, DEFAULT_FILES_STORAGE_PATH);
> }

What do you think?

Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/FileSystemFileService.java]: Done

**Comment by nictas** [comment]
I think we decided against implementing this sort of clean-up, since a problem might occur if you delete a directory that was just about to get used.

**Comment by nictas** [comment]
Apparently, this pull request makes sure that directories are only deleted if the space was deleted. This prevents the problem described in my previous comment.

I'm reopening it.

**Comment by nictas** [comment]
Apparently, this pull request makes sure that directories are only deleted if the space was deleted. This prevents the problem described in my previous comment.

I'm reopening it.

**Comment by nictas** [comment]
Apparently, this pull request makes sure that directories are only deleted if the space was deleted. This prevents the problem described in my previous comment.

I'm reopening it.

File: com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/AbstractFileService.java (hunk 2 above)
Comment (nictas) [reply com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/AbstractFileService.java]: Why not "deleteBySpace"? "deleteSpaceLeftovers" adds some context coming from the clean-up job's perspective, which should not be in the FileService.

**Comment by nictas** [comment]
Apparently, this pull request makes sure that directories are only deleted if the space was deleted. This prevents the problem described in my previous comment.

I'm reopening it.

**Comment by nictas** [comment]
Apparently, this pull request makes sure that directories are only deleted if the space was deleted. This prevents the problem described in my previous comment.

I'm reopening it.

**Comment by enchobelezirev** [comment]
Момчета, лошо ми е ббрее!!!!

**Pull request merged**